xonsh/shell.py

```diff
@@ -87,7 +89,12 @@ def __init__(self, ctx=None, shell_type=None, config=None, rc=None,
         self._init_environ(ctx, config, rc,
                            kwargs.get('scriptcache', True),
                            kwargs.get('cacheall', False))
+
         env = builtins.__xonsh_env__
+        # build history backend before creating shell
+        builtins.__xonsh_history__ = xhm.construct_history(
+            env=env.detype(), ts=[time.time(), None], locked=True)
```

Comment:
I moved it here so that history backend is created after loading xonshrc and knows which one need to be created (but before shell got created). Should I achieve this through other ways?